tests/integration/test_wazuh_db/test_wazuh_db.py

```diff
@@ -56,8 +58,47 @@ def regex_match(regex, string):
     return re.match(regex, string)
 
 
+def wait_range_checksum_avoided(line):
+    """Callback function to wait until the manager avoided the checksum calculus by using the last saved one."""
+    if 'range checksum avoided' in line:
+        return line
+    return None
+
+
+def wait_range_checksum_calculated(line):
+    """Callback function to wait until the manager calculates the new checksum."""
+    if 'range checksum: Time: ' in line:
+        return line
+    return None
+
+
 # Tests
 
+@pytest.fixture(scope="function")
+def prepare_range_checksum_data():
+    truncate_file(LOG_FILE_PATH)
+
+    command = """agent 003 syscheck save2 {\"path\":\"/home/test/file1\",\"timestamp\":1575421292,
+                 \"attributes\":{\"type\":\"file\",\"size\":0,\"perm\":\"rw-r--r--\",\"uid\":\"0\",\"gid\":\"0\",
+                 \"user_name\":\"root\",\"group_name\":\"root\",\"inode\":16879,\"mtime\":1575421292,
+                 \"hash_md5\":\"d41d8cd98f00b204e9800998ecf8427e\",
+                 \"hash_sha1\":\"da39a3ee5e6b4b0d3255bfef95601890afd80709\",
+                 \"hash_sha256\":\"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\",
+                 \"checksum\":\"f65b9f66c5ef257a7566b98e862732640d502b6f\"}}"""
+    receiver_sockets[0].send(command, size=True)
+    response = receiver_sockets[0].receive(size=True).decode()
+
+    command = """agent 003 syscheck save2 {\"path\":\"/home/test/file2\",\"timestamp\":1575421292,
+                 \"attributes\":{\"type\":\"file\",\"size\":0,\"perm\":\"rw-r--r--\",\"uid\":\"0\",\"gid\":\"0\",
+                 \"user_name\":\"root\",\"group_name\":\"root\",\"inode\":16879,\"mtime\":1575421292,
+                 \"hash_md5\":\"d41d8cd98f00b204e9800998ecf8427e\",
+                 \"hash_sha1\":\"da39a3ee5e6b4b0d3255bfef95601890afd80709\",
+                 \"hash_sha256\":\"e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\",
+                 \"checksum\":\"f65b9f66c5ef257a7566b98e862732640d502b6f\"}}"""
+    receiver_sockets[0].send(command, size=True)
+    response = receiver_sockets[0].receive(size=True).decode()
```

Comment:
I think these commands should be defined using the `json` module and a little string logic.
```suggestion
file_path = '/home/test/file1'
command_body = {'path': f"{file_path}",'timestamp':1575421292, 'attributes':{'type':'file','size':0,'perm':'rw-r--r--','uid':'0','gid':'0', 'user_name':'root','group_name':'root','inode':16879,'mtime':1575421292, 'hash_md5':'d41d8cd98f00b204e9800998ecf8427e', 'hash_sha1':'da39a3ee5e6b4b0d3255bfef95601890afd80709', 'hash_sha256':'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855', 'checksum':'f65b9f66c5ef257a7566b98e862732640d502b6f'}}
command_header = 'agent 003 syscheck save2 '

command = command_header + json.dumps(command_body)

receiver_sockets[0].send(command, size=True)
response = receiver_sockets[0].receive(size=True).decode()

command_body['path'] = '/home/test/file2'
command = command_header + json.dumps(command_body)

receiver_sockets[0].send(command, size=True)
response = receiver_sockets[0].receive(size=True).decode()
```

Reply:
Is the response needed?

Reply:
FIxed